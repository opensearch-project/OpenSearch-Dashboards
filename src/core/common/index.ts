/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

export { ImageType, ColorScheme, getLogos } from './logos';
export type { Logos } from './types';
