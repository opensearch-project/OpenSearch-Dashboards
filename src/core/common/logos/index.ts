/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

// ToDo: Extend this to handle everything related to white-labelling

export { getLogos } from './get_logos';
export { ImageType, ColorScheme } from './constants';
