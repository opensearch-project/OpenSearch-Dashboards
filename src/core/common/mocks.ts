/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

export { getLogosMock } from './logos/get_logos.mock';
