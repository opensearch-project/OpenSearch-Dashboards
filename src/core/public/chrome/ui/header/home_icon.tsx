/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import React from 'react';
import { EuiIcon, IconSize } from '@elastic/eui';
import { ChromeBranding } from '../../chrome_service';
import type { Logos } from '../../../../common/types';

interface Props {
  branding: ChromeBranding;
  logos: Logos;
}

/**
 * Use branding configurations to render the header mark on the nav bar.
 */
export const HomeIcon = ({ branding, logos }: Props) => {
  const { applicationTitle = 'opensearch dashboards', useExpandedHeader = true } = branding;

  const { url: markURL, type: markType } = logos.Mark;

  let markIcon = markURL;
  let testSubj = `${markType}Mark`;
  // Marks look better at the large size
  let markIconSize: IconSize = 'l';

  // If no custom branded mark was set, use `home` icon only for expanded headers
  if (markType !== 'custom' && useExpandedHeader) {
    markIcon = 'home';
    testSubj = 'homeIcon';
    // Home icon should be medium to fit in with other icons
    markIconSize = 'm';
  }

  const alt = `${applicationTitle} home`;

  return (
    <EuiIcon
      data-test-subj={testSubj}
      data-test-image-url={markIcon}
      type={markIcon}
      title={alt}
      size={markIconSize}
      className="logoImage"
    />
  );
};
