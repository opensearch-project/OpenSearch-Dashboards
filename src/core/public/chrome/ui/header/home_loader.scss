.header__homeLoaderNavButton .euiHeaderSectionItemButton__content {
  // avoid layout shift on smallest breakpoint if logo and loading indicator are different sizes
  min-width: 24px;
  display: grid;
  grid-template-columns: 1fr;
  grid-template-rows: 1fr;

  & .loaderContainer,
  & .homeIconContainer {
    grid-area: 1 / 1;
    align-self: center;
    justify-self: center;
  }
}
