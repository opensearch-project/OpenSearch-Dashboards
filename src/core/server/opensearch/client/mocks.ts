/*
 * SPDX-License-Identifier: Apache-2.0
 *
 * The OpenSearch Contributors require contributions made to
 * this file be licensed under the Apache-2.0 license or a
 * compatible open source license.
 *
 * Any modifications Copyright OpenSearch Contributors. See
 * GitHub history for details.
 */

/*
 * Licensed to Elasticsearch B.V. under one or more contributor
 * license agreements. See the NOTICE file distributed with
 * this work for additional information regarding copyright
 * ownership. Elasticsearch B.V. licenses this file to you under
 * the Apache License, Version 2.0 (the "License"); you may
 * not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *    http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing,
 * software distributed under the License is distributed on an
 * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
 * KIND, either express or implied.  See the License for the
 * specific language governing permissions and limitations
 * under the License.
 */

import { Client, ApiResponse } from '@opensearch-project/opensearch';
import { TransportRequestPromise } from '@opensearch-project/opensearch/lib/Transport';
import { OpenSearchClient } from './types';
import { ICustomClusterClient } from './cluster_client';

const createInternalClientMock = (withLongNumeralsSupport = false): DeeplyMockedKeys<Client> => {
  // we mimic 'reflection' on a concrete instance of the client to generate the mocked functions.
  const client = new Client({
    node: 'http://localhost',
    // @ts-expect-error - ToDo: Remove ignoring after https://github.com/opensearch-project/opensearch-js/pull/598 is included in a release
    enableLongNumeralSupport: withLongNumeralsSupport,
  }) as any;

  const omittedProps = [
    '_events',
    '_eventsCount',
    '_maxListeners',
    'constructor',
    'name',
    'serializer',
    'connectionPool',
    'transport',
    'helpers',
  ];

  const getAllPropertyDescriptors = (obj: Record<string, any>) => {
    const descriptors = Object.entries(Object.getOwnPropertyDescriptors(obj));
    let prototype = Object.getPrototypeOf(obj);
    while (prototype != null && prototype !== Object.prototype) {
      descriptors.push(...Object.entries(Object.getOwnPropertyDescriptors(prototype)));
      prototype = Object.getPrototypeOf(prototype);
    }
    return descriptors;
  };

  const mockify = (obj: Record<string, any>, omitted: string[] = []) => {
    // the @opensearch-project/opensearch::Client uses prototypical inheritance
    // so we have to crawl up the prototype chain and get all descriptors
    // to find everything that we should be mocking
    const descriptors = getAllPropertyDescriptors(obj);
    descriptors
      .filter(([key]) => !omitted.includes(key))
      .forEach(([key, descriptor]) => {
        if (typeof descriptor.value === 'function') {
          obj[key] = jest.fn(() => createSuccessTransportRequestPromise({}));
        } else if (typeof obj[key] === 'object' && obj[key] != null) {
          mockify(obj[key], omitted);
        }
      });
  };

  mockify(client, omittedProps);

  client.close = jest.fn().mockReturnValue(Promise.resolve());
  client.child = jest
    .fn()
    .mockImplementation(() => createInternalClientMock(withLongNumeralsSupport));

  const mockGetter = (obj: Record<string, any>, propertyName: string) => {
    Object.defineProperty(obj, propertyName, {
      configurable: true,
      enumerable: false,
      get: () => jest.fn(),
      set: undefined,
    });
  };

  // `on`, `off`, and `once` are properties without a setter.
  // We can't `client.on = jest.fn()` because the following error will be thrown:
  // TypeError: Cannot set property on of #<Client> which has only a getter
  mockGetter(client, 'on');
  mockGetter(client, 'off');
  mockGetter(client, 'once');
  client.transport = {
    request: jest.fn(),
  };

  return client as DeeplyMockedKeys<Client>;
};

export type OpenSearchClientMock = DeeplyMockedKeys<OpenSearchClient>;

const createClientMock = (withLongNumeralsSupport = false): OpenSearchClientMock =>
  (createInternalClientMock(withLongNumeralsSupport) as unknown) as OpenSearchClientMock;

export interface ScopedClusterClientMock {
  asInternalUser: OpenSearchClientMock;
  asCurrentUser: OpenSearchClientMock;
  asInternalUserWithLongNumeralsSupport: OpenSearchClientMock;
  asCurrentUserWithLongNumeralsSupport: OpenSearchClientMock;
}

const createScopedClusterClientMock = () => {
  const mock: ScopedClusterClientMock = {
    asInternalUser: createClientMock(),
    asCurrentUser: createClientMock(),
    asInternalUserWithLongNumeralsSupport: createClientMock(true),
    asCurrentUserWithLongNumeralsSupport: createClientMock(true),
  };

  return mock;
};

export interface ClusterClientMock {
  asInternalUser: OpenSearchClientMock;
  asInternalUserWithLongNumeralsSupport: OpenSearchClientMock;
  asScoped: jest.MockedFunction<() => ScopedClusterClientMock>;
}

const createClusterClientMock = () => {
  const mock: ClusterClientMock = {
    asInternalUser: createClientMock(),
    asInternalUserWithLongNumeralsSupport: createClientMock(true),
    asScoped: jest.fn(),
  };

  mock.asScoped.mockReturnValue(createScopedClusterClientMock());

  return mock;
};

export type CustomClusterClientMock = jest.Mocked<ICustomClusterClient> & ClusterClientMock;

const createCustomClusterClientMock = () => {
  const mock: CustomClusterClientMock = {
    asInternalUser: createClientMock(),
    asInternalUserWithLongNumeralsSupport: createClientMock(true),
    asScoped: jest.fn(),
    close: jest.fn(),
  };

  mock.asScoped.mockReturnValue(createScopedClusterClientMock());
  mock.close.mockReturnValue(Promise.resolve());

  return mock;
};

export type MockedTransportRequestPromise<T> = TransportRequestPromise<T> & {
  abort: jest.MockedFunction<() => undefined>;
};

const createSuccessTransportRequestPromise = <T>(
  body: T,
  { statusCode = 200 }: { statusCode?: number } = {}
): MockedTransportRequestPromise<ApiResponse<T>> => {
  const response = createApiResponse({ body, statusCode });
  const promise = Promise.resolve(response);
  (promise as MockedTransportRequestPromise<ApiResponse<T>>).abort = jest.fn();

  return promise as MockedTransportRequestPromise<ApiResponse<T>>;
};

const createErrorTransportRequestPromise = (err: any): MockedTransportRequestPromise<never> => {
  const promise = Promise.reject(err);
  (promise as MockedTransportRequestPromise<never>).abort = jest.fn();
  return promise as MockedTransportRequestPromise<never>;
};

function createApiResponse<TResponse = Record<string, any>>(
  opts: Partial<ApiResponse> = {}
): ApiResponse<TResponse> {
  return {
    body: {} as any,
    statusCode: 200,
    headers: {},
    warnings: [],
    meta: {} as any,
    ...opts,
  };
}

export const opensearchClientMock = {
  createClusterClient: createClusterClientMock,
  createCustomClusterClient: createCustomClusterClientMock,
  createScopedClusterClient: createScopedClusterClientMock,
  createOpenSearchClient: createClientMock,
  createInternalClient: createInternalClientMock,
  createSuccessTransportRequestPromise,
  createErrorTransportRequestPromise,
  createApiResponse,
};
