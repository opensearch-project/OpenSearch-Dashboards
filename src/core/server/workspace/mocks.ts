/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import { IWorkspaceService } from '.';
import { InternalWorkspaceServiceSetup, InternalWorkspaceServiceStart } from './workspace_service';

const createWorkspaceServiceMock = () => {
  const mocked: jest.Mocked<IWorkspaceService> = {
    setup: jest.fn().mockReturnValue(createInternalSetupContractMock()),
    start: jest.fn().mockReturnValue({}),
    stop: jest.fn(),
  };

  return mocked;
};

const createInternalSetupContractMock = () => {
  const mocked: jest.Mocked<InternalWorkspaceServiceSetup> = {
    isWorkspaceEnabled: jest.fn(),
  };

  return mocked;
};
const createSetupContractMock = createInternalSetupContractMock;

const createInternalStartContractMock = () => {
  const mocked: jest.Mocked<InternalWorkspaceServiceStart> = {
    isWorkspaceEnabled: jest.fn(),
  };

  return mocked;
};
const createStartContractMock = createInternalStartContractMock;

export const workspaceServiceMock = {
  create: createWorkspaceServiceMock,
  createInternalSetupContract: createInternalSetupContractMock,
  createInternalStartContract: createInternalStartContractMock,
  createSetupContract: createSetupContractMock,
  createStartContract: createStartContractMock,
};
