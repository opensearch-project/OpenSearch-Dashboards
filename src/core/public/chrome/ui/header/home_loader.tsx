/*
 * SPDX-License-Identifier: Apache-2.0
 *
 * The OpenSearch Contributors require contributions made to
 * this file be licensed under the Apache-2.0 license or a
 * compatible open source license.
 *
 * Any modifications Copyright OpenSearch Contributors. See
 * GitHub history for details.
 */

/*
 * Licensed to Elasticsearch B.V. under one or more contributor
 * license agreements. See the NOTICE file distributed with
 * this work for additional information regarding copyright
 * ownership. Elasticsearch B.V. licenses this file to you under
 * the Apache License, Version 2.0 (the "License"); you may
 * not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *    http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing,
 * software distributed under the License is distributed on an
 * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
 * KIND, either express or implied.  See the License for the
 * specific language governing permissions and limitations
 * under the License.
 */

import './home_loader.scss';

import { i18n } from '@osd/i18n';
import React from 'react';
import useObservable from 'react-use/lib/useObservable';
import { Observable } from 'rxjs';
import Url from 'url';
import { EuiHeaderSectionItemButton } from '@elastic/eui';
import { ChromeNavLink } from '../..';
import { ChromeBranding } from '../../chrome_service';
import { LoadingIndicator } from '../loading_indicator';
import { HomeIcon } from './home_icon';

function findClosestAnchor(element: HTMLElement): HTMLAnchorElement | void {
  let current = element;
  while (current) {
    if (current.tagName === 'A') {
      return current as HTMLAnchorElement;
    }

    if (!current.parentElement || current.parentElement === document.body) {
      return undefined;
    }

    current = current.parentElement;
  }
}

function onClick(
  event: React.MouseEvent<HTMLAnchorElement>,
  forceNavigation: boolean,
  navLinks: ChromeNavLink[],
  navigateToApp: (appId: string) => void
) {
  const anchor = findClosestAnchor((event as any).nativeEvent.target);
  if (!anchor) {
    return;
  }

  const navLink = navLinks.find((item) => item.href === anchor.href);
  if (navLink && navLink.disabled) {
    event.preventDefault();
    return;
  }

  if (event.isDefaultPrevented() || event.altKey || event.metaKey || event.ctrlKey) {
    return;
  }

  if (forceNavigation) {
    const toParsed = Url.parse(anchor.href);
    const fromParsed = Url.parse(document.location.href);
    const sameProto = toParsed.protocol === fromParsed.protocol;
    const sameHost = toParsed.host === fromParsed.host;
    const samePath = toParsed.path === fromParsed.path;

    if (sameProto && sameHost && samePath) {
      if (toParsed.hash) {
        document.location.reload();
      }

      // event.preventDefault() keeps the browser from seeing the new url as an update
      // and even setting window.location does not mimic that behavior, so instead
      // we use stopPropagation() to prevent angular from seeing the click and
      // starting a digest cycle/attempting to handle it in the router.
      event.stopPropagation();
    }
  } else {
    navigateToApp('home');
    event.preventDefault();
  }
}

interface Props {
  href: string;
  navLinks$: Observable<ChromeNavLink[]>;
  forceNavigation$: Observable<boolean>;
  loadingCount$: Observable<number>;
  navigateToApp: (appId: string) => void;
  branding: ChromeBranding;
}

export function HomeLoader({ href, navigateToApp, branding, ...observables }: Props) {
  const forceNavigation = useObservable(observables.forceNavigation$, false);
  const navLinks = useObservable(observables.navLinks$, []);
  const loadingCount = useObservable(observables.loadingCount$, 0);
  const label = i18n.translate('core.ui.chrome.headerGlobalNav.goHomePageIconAriaLabel', {
    defaultMessage: 'Go to home page',
  });

  return (
    <EuiHeaderSectionItemButton
      className="header__homeLoaderNavButton"
      data-test-subj="homeLoader"
      onClick={(e: React.MouseEvent<HTMLAnchorElement>) =>
        onClick(e, forceNavigation, navLinks, navigateToApp)
      }
      href={href}
      aria-label={label}
      title={label}
    >
      {!(loadingCount > 0) && (
        <div className="homeIconContainer">
          <HomeIcon {...branding} />
        </div>
      )}
      <div className="loaderContainer">
        <LoadingIndicator loadingCount$={observables.loadingCount$} />
      </div>
    </EuiHeaderSectionItemButton>
  );
}
