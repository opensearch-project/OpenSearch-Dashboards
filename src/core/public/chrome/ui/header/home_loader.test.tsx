/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import React from 'react';
import { BehaviorSubject } from 'rxjs';
import { getLogosMock } from '../../../../common/mocks';
import { mountWithIntl, shallowWithIntl } from 'test_utils/enzyme_helpers';
import { HomeLoader } from './home_loader';
import { LoadingIndicator } from '../loading_indicator';
import { EuiToolTip } from '@elastic/eui';
import { HomeIcon } from './home_icon';

const mockProps = () => ({
  href: '/',
  navLinks$: new BehaviorSubject([]),
  forceNavigation$: new BehaviorSubject(false),
  navigateToApp: jest.fn(),
  logos: getLogosMock.default,
  branding: {},
});

describe('Home loader', () => {
  it('shows the loading indicator if loading count > 0', () => {
    const props = {
      ...mockProps(),
      loadingCount$: new BehaviorSubject(1),
    };
    const component = shallowWithIntl(<HomeLoader {...props} />);
    const loadingIndicator = component.find(LoadingIndicator);
    expect(loadingIndicator.exists()).toBeTruthy();
    expect(loadingIndicator.prop('loadingCount$')).toEqual(props.loadingCount$);
    expect(component).toMatchSnapshot();
  });
  it('displays a EuiToolTip', () => {
    const props = {
      ...mockProps(),
      loadingCount$: new BehaviorSubject(0),
    };
    const component = shallowWithIntl(<HomeLoader {...props} />);
    const toolTip = component.find(EuiToolTip);
    expect(toolTip.exists()).toBeTruthy();
    expect(component).toMatchSnapshot();
  });
  describe('onClick', () => {
    it('directs to home page', () => {
      const props = {
        ...mockProps(),
        loadingCount$: new BehaviorSubject(0),
      };
      const component = mountWithIntl(<HomeLoader {...props} />);
      const homeIcon = component.find(HomeIcon);
      expect(homeIcon.exists()).toBeTruthy();
      homeIcon.simulate('click');
      expect(props.navigateToApp).toHaveBeenCalledTimes(1);
      expect(props.navigateToApp).toHaveBeenCalledWith('home');
    });
  });
});
