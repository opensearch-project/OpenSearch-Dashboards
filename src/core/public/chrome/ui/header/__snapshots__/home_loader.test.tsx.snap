// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home loader displays a EuiToolTip 1`] = `
<EuiToolTip
  content="Go to home page"
  delay="long"
  position="top"
>
  <EuiHeaderSectionItemButton
    aria-label="Go to home page"
    className="header__homeLoaderNavButton"
    data-test-subj="homeLoader"
    href="/"
    onClick={[Function]}
  >
    <div
      className="homeIconContainer"
    >
      <HomeIcon
        branding={Object {}}
        logos={
          Object {
            "AnimatedMark": Object {
              "dark": Object {
                "type": "default",
                "url": "/ui/logos/opensearch_spinner_on_dark.svg",
              },
              "light": Object {
                "type": "default",
                "url": "/ui/logos/opensearch_spinner_on_light.svg",
              },
              "type": "default",
              "url": "/ui/logos/opensearch_spinner_on_light.svg",
            },
            "Application": Object {
              "dark": Object {
                "type": "default",
                "url": "/ui/logos/opensearch_dashboards_on_dark.svg",
              },
              "light": Object {
                "type": "default",
                "url": "/ui/logos/opensearch_dashboards_on_light.svg",
              },
              "type": "default",
              "url": "/ui/logos/opensearch_dashboards_on_light.svg",
            },
            "CenterMark": Object {
              "dark": Object {
                "type": "default",
                "url": "/ui/logos/opensearch_center_mark_on_dark.svg",
              },
              "light": Object {
                "type": "default",
                "url": "/ui/logos/opensearch_center_mark_on_light.svg",
              },
              "type": "default",
              "url": "/ui/logos/opensearch_center_mark_on_light.svg",
            },
            "Mark": Object {
              "dark": Object {
                "type": "default",
                "url": "/ui/logos/opensearch_mark_on_dark.svg",
              },
              "light": Object {
                "type": "default",
                "url": "/ui/logos/opensearch_mark_on_light.svg",
              },
              "type": "default",
              "url": "/ui/logos/opensearch_mark_on_light.svg",
            },
            "OpenSearch": Object {
              "dark": Object {
                "type": "default",
                "url": "/ui/logos/opensearch_on_dark.svg",
              },
              "light": Object {
                "type": "default",
                "url": "/ui/logos/opensearch_on_light.svg",
              },
              "type": "default",
              "url": "/ui/logos/opensearch_on_light.svg",
            },
            "colorScheme": "light",
          }
        }
      />
    </div>
    <div
      className="loaderContainer"
    >
      <LoadingIndicator
        loadingCount$={
          BehaviorSubject {
            "_isScalar": false,
            "_value": 0,
            "closed": false,
            "hasError": false,
            "isStopped": false,
            "observers": Array [],
            "thrownError": null,
          }
        }
        showAsBar={false}
      />
    </div>
  </EuiHeaderSectionItemButton>
</EuiToolTip>
`;

exports[`Home loader shows the loading indicator if loading count > 0 1`] = `
<EuiToolTip
  content="Go to home page"
  delay="long"
  position="top"
>
  <EuiHeaderSectionItemButton
    aria-label="Go to home page"
    className="header__homeLoaderNavButton"
    data-test-subj="homeLoader"
    href="/"
    onClick={[Function]}
  >
    <div
      className="homeIconContainer"
    >
      <HomeIcon
        branding={Object {}}
        logos={
          Object {
            "AnimatedMark": Object {
              "dark": Object {
                "type": "default",
                "url": "/ui/logos/opensearch_spinner_on_dark.svg",
              },
              "light": Object {
                "type": "default",
                "url": "/ui/logos/opensearch_spinner_on_light.svg",
              },
              "type": "default",
              "url": "/ui/logos/opensearch_spinner_on_light.svg",
            },
            "Application": Object {
              "dark": Object {
                "type": "default",
                "url": "/ui/logos/opensearch_dashboards_on_dark.svg",
              },
              "light": Object {
                "type": "default",
                "url": "/ui/logos/opensearch_dashboards_on_light.svg",
              },
              "type": "default",
              "url": "/ui/logos/opensearch_dashboards_on_light.svg",
            },
            "CenterMark": Object {
              "dark": Object {
                "type": "default",
                "url": "/ui/logos/opensearch_center_mark_on_dark.svg",
              },
              "light": Object {
                "type": "default",
                "url": "/ui/logos/opensearch_center_mark_on_light.svg",
              },
              "type": "default",
              "url": "/ui/logos/opensearch_center_mark_on_light.svg",
            },
            "Mark": Object {
              "dark": Object {
                "type": "default",
                "url": "/ui/logos/opensearch_mark_on_dark.svg",
              },
              "light": Object {
                "type": "default",
                "url": "/ui/logos/opensearch_mark_on_light.svg",
              },
              "type": "default",
              "url": "/ui/logos/opensearch_mark_on_light.svg",
            },
            "OpenSearch": Object {
              "dark": Object {
                "type": "default",
                "url": "/ui/logos/opensearch_on_dark.svg",
              },
              "light": Object {
                "type": "default",
                "url": "/ui/logos/opensearch_on_light.svg",
              },
              "type": "default",
              "url": "/ui/logos/opensearch_on_light.svg",
            },
            "colorScheme": "light",
          }
        }
      />
    </div>
    <div
      className="loaderContainer"
    >
      <LoadingIndicator
        loadingCount$={
          BehaviorSubject {
            "_isScalar": false,
            "_value": 1,
            "closed": false,
            "hasError": false,
            "isStopped": false,
            "observers": Array [],
            "thrownError": null,
          }
        }
        showAsBar={false}
      />
    </div>
  </EuiHeaderSectionItemButton>
</EuiToolTip>
`;
